contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.3.0--4.4.0.sql

```diff
@@ -10566,6 +10566,50 @@ CALL sys.babelfish_alter_default_privilege_on_schema();
 -- Drop this procedure after it gets executed once.
 DROP PROCEDURE sys.babelfish_alter_default_privilege_on_schema();
 
+-- To handle the typmod values for procedure and function
+UPDATE pg_proc p1
+SET probin = 
+    CASE 
+        WHEN p1.probin::jsonb->'typmod_array' = '[]'::jsonb THEN p1.probin
+        ELSE (
+            SELECT (
+                jsonb_set(
+                    p1.probin::jsonb,
+                    '{typmod_array}',
+                    to_jsonb(
+                        (
+                            SELECT jsonb_agg(
+                                CASE
+                                    WHEN p2.prokind = 'p' OR (p2.prokind = 'f' AND p2.proallargtypes IS NULL AND p2.proargtypes[typ_index-1] IS NOT NULL) THEN
+                                        CASE
+                                            WHEN typmod::text = '-8000' AND p2.proargtypes[typ_index-1]::regtype in ('sys.varchar', 'sys.nvarchar', 'sys.varbinary') THEN '-1'
```

Comment:
Do we really need to explicitly cast typmod to text?

Reply:
Removed